ts/session/apis/snode_api/swarmPolling.ts

```diff
@@ -105,6 +113,33 @@ function entryToKey(entry: GroupPollingEntry) {
   return entry.pubkey.key;
 }
 
+function mergeMultipleRetrieveResults(
+  results: RetrieveMessagesResultsBatched
+): RetrieveMessagesResultsMergedBatched {
+  const mapped: Map<SnodeNamespaces, Map<string, RetrieveMessageItem>> = new Map();
+  for (let resultIndex = 0; resultIndex < results.length; resultIndex++) {
+    const result = results[resultIndex];
+    if (!mapped.has(result.namespace)) {
+      mapped.set(result.namespace, new Map());
+    }
+    if (result.messages.messages) {
+      for (let msgIndex = 0; msgIndex < result.messages.messages.length; msgIndex++) {
+        const msg = result.messages.messages[msgIndex];
+        if (!mapped.get(result.namespace)!.has(msg.hash)) {
+          mapped.get(result.namespace)!.set(msg.hash, msg);
+        }
+      }
+    }
+  }
+
+  // Convert the merged map back to an array
+  return Array.from(mapped.entries()).map(([namespace, messagesMap]) => ({
+    code: 200, // Assuming success code, adjust as needed
```

Comment:
So do responses that are not 200 never get this far in the receiving pipeline? When and how would one 'adjust' this status code. I do see that `RetrieveRequestResult` which is part of `RetrieveMessagesResultsBatched` has `code` as a member. Can we not use that instead of hardcoding this value?

Reply:
that's correct.

Reply:
amended my commit